plugins/inputs/bond/bond.go

```diff
@@ -118,10 +159,98 @@ func (bond *Bond) gatherBondPart(bondName string, rawFile string, acc telegraf.A
 	return fmt.Errorf("Couldn't find status info for '%s' ", bondName)
 }
 
+func (bond *Bond) readSysFiles(bondDir string) (sysFiles, error) {
+	/*
+		Files we may need
+		bonding/mode
+		bonding/slaves
+		bonding/ad_num_ports
+
+		We load files here first to allow for easier testing
+	*/
+	var output sysFiles
+
+	file, err := os.ReadFile(bondDir + "/bonding/mode")
+	if err != nil {
+		return sysFiles{}, fmt.Errorf("error inspecting %q interface: %v", bondDir+"/bonding/mode", err)
+	}
+	output.ModeFile = strings.TrimSpace(string(file))
+	file, err = os.ReadFile(bondDir + "/bonding/slaves")
+	if err != nil {
+		return sysFiles{}, fmt.Errorf("error inspecting %q interface: %v", bondDir+"/bonding/slaves", err)
+	}
+	output.SlaveFile = strings.TrimSpace(string(file))
+	if bond.BondType == "IEEE 802.3ad Dynamic link aggregation" {
+		file, err = os.ReadFile(bondDir + "/bonding/ad_num_ports")
+		if err != nil {
+			return sysFiles{}, fmt.Errorf("error inspecting %q interface: %v", bondDir+"/bonding/ad_num_ports", err)
+		}
+		output.ADPortsFile = strings.TrimSpace(string(file))
+	}
+	return output, nil
+}
+
+func (bond *Bond) gatherSysDetails(bondName string, files sysFiles, acc telegraf.Accumulator) {
+	var slaves []string
+	var adPortCount int
+
+	/*
+		To start with, we get the bond operating mode
+		This file should only have one line in it
+		so we can simply scan the one line
+	*/
+	mode := strings.TrimSpace(strings.Split(files.ModeFile, " ")[0])
+
+	tags := map[string]string{
+		"bond": bondName,
+		"mode": mode,
+	}
+
+	/*
+		Next we collect the number of bond slaves the system expects
+		This file should only have one line in it
+		so we can simply scan the one line
+	*/
+	slavesTmp := strings.Split(files.SlaveFile, " ")
+	for _, slave := range slavesTmp {
+		if slave != "" {
+			slaves = append(slaves, slave)
+		}
+	}
+	// bond.Log.Debugf("Slaves: %v", slaves)
```

Comment:
Please either keep the debug message (i.e. uncomment that line) or delete the commented line. Both is fine with me.